src/components/cluster-deployments/EditClusterWizard.tsx

```diff
@@ -0,0 +1,53 @@
+import * as React from 'react';
+import { RouteComponentProps } from 'react-router';
+import { useK8sWatchResource } from '@openshift-console/dynamic-plugin-sdk/api';
+import { AgentClusterInstallK8sResource, ClusterDeploymentK8sResource } from '../types';
+import { AgentClusterInstallKind, ClusterDeploymentKind } from '../../kind';
+import ClusterDeploymentWizard from './ClusterDeploymentWizard';
+
+const EditClusterWizard: React.FC<RouteComponentProps<{ ns: string; name: string }>> = ({
+  match,
+  history,
+}) => {
+  const namespace = match.params?.ns;
+  const name = match.params?.name;
+
+  const [clusterDeployment, clusterDeploymentLoaded, clusterDeploymentError] =
+    useK8sWatchResource<ClusterDeploymentK8sResource>({
+      kind: ClusterDeploymentKind,
+      name,
+      namespace,
+      namespaced: true,
+      isList: false,
+    });
+
+  // it is ok if missing
+  const [agentClusterInstall] = useK8sWatchResource<AgentClusterInstallK8sResource>({
+    kind: AgentClusterInstallKind,
+    name: clusterDeployment?.spec?.clusterInstallRef?.name || '___missing__',
+    namespace,
+    namespaced: true,
+    isList: false,
+  });
+
```

Comment:
if there is no `clusterInstallRef` then a better would be not do any remote request. For that you can just pass undefined as param to `useK8sWatchResource`

```suggestion
  const refName = clusterDeployment?.spec?.clusterInstallRef?.name;
  const [agentClusterInstall] = useK8sWatchResource<AgentClusterInstallK8sResource>(refName ? {
    kind: AgentClusterInstallKind,
    name: refName,
    namespace,
    namespaced: true,
    isList: false,
  } : undefined);

```

Reply:
thanks, done